src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,51 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $calendar-gap;
+  width: $calendar-width;
+  padding: $calendar-padding;
+
+  $start-day:
+    "mon" 0,
+    "tue" ($cell-size + $calendar-gap),
+    "wed" ($cell-size + $calendar-gap) * 2,
+    "thu" ($cell-size + $calendar-gap) * 3,
+    "fri" ($cell-size + $calendar-gap) * 4,
+    "sat" ($cell-size + $calendar-gap) * 5,
+    "sun" ($cell-size + $calendar-gap) * 6;
+
+  @each $day, $margin in $start-day {
+    &--start-day--#{$day} &__cell:first-child {
+      margin-left: $margin;
+    }
+  }
+
+  @for $i from 28 through 30 {
+    &--month-length--#{$i} &__cell:nth-child(n + #{$i + 1}) {
+      display: none;
+    }
+  }
+
+  &__cell {
+    @include square($cell-size);
+    box-sizing: border-box;
+    border: 1px solid black;
+    background-color: #eee;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    @for $count from 1 through 31 {
+      &:nth-child(#{$count})::before {
+        content: "#{$count}";
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+      transition: 0.5s;
```

Comment:
Move transition under general selector, so smooth movement works both ways.
Also, add parameters that you transition.